linkerd/meshtls/boring/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+#![deny(warnings, rust_2018_idioms)]
+#![forbid(unsafe_code)]
+
+//! This crate provides an implementation of _meshtls_ backed by `boringssl` (as
+//! provided by <https://github.com/cloudflare/boring>).
+//!
+//! There are several caveats with the current implementation:
+//!
+//! In its current form, this crate is compatible with the `meshtls-rustls`
+//! implementation, which requires of ECDSA-P256-SHA256 keys & signature
+//! algorithms. This crate doesn't actually constrain the algorithms beyond the
+//! Mozilla's 'intermediate' (v5) [defaults][defaults]. But, the goal for
+//! supporting `boring` is to provide a FIPS 140-2 compliant mode. There's a
+//! [PR][fips-pr] that implements this, but code changes will likely be required
+//! to enable this once it's merged/released.
+//!
+//! A new SSL context is created for each connection. This is probably
+//! unnecessary, but it's simpler for now. We can revisit this if needed.
```

Comment:
not a blocker at all, just curious: how much do we know about what this actually *means*? e.g. do we know what a "SSL context" in `boring` consists of, and what are the implications of creating them?

Reply:
Some allocations + everything in the connector/acceptor constructors (i.e. trust anchors, private key, certs).